ocaml/xapi/xapi_sr.ml

```diff
@@ -240,12 +240,15 @@ let create  ~__context ~host ~device_config ~(physical_size:int64) ~name_label ~
   end;
   Helpers.call_api_functions ~__context
     (fun rpc session_id ->
-       List.iter
-         (fun self ->
-            try
-              Client.PBD.plug ~rpc ~session_id ~self
-            with e -> warn "Could not plug PBD '%s': %s" (Db.PBD.get_uuid ~__context ~self) (Printexc.to_string e))
-         pbds);
+       let tasks = List.fold_left (fun tasks self ->
+           try
+             let task = Client.Async.PBD.plug ~rpc ~session_id ~self in
+             task :: tasks
+           with e ->
+             warn "Could not plug PBD '%s': %s" (Db.PBD.get_uuid ~__context ~self) (Printexc.to_string e);
```

Comment:
This almost certainly does very little - all you're checking for is that the async call was successfully started, not that it completed successfully. I think you're probably better off just replacing the whole fold_left thing with just

```ocaml
let tasks = List.map (fun self -> Client.Async.PBD.plug ~rpc ~session_id ~self) pbds in
Tasks.wait_for_all ~rpc ~session_id ~tasks
```

Reply:
Hmm, will map check that the async call was completed successfully?

Reply:
No, but `wait_for_all` will wait for the completion

Reply:
yes.  So what the difference by using map instead of left_fold?

Reply:
It's easier to read. The comment was more about the fact that the `try/with` in the fold is not needed because it does nothing, and in the end you are using a fold to just to a map, so you might as well use a map and make the code more readable

Reply:
If anything it would become something like:
```ocaml
try
   let tasks = List.map (fun self -> Client.Async.PBD.plug ~rpc ~session_id ~self) pbds in
   Tasks.wait_for_all ~rpc ~session_id ~tasks
with err ->
   warn "Could not plug some PBDs: %s" (Printexc.to_string err);
```
and I am not even sure that the `err` that you get there is informative at all.

Reply:
Ok. Understand your point. Thanks @mseri @jonludlam . 
If so, I would just use map as @jonludlam  suggests. Will update it later.   Thanks you.